### ISSUE TITLE
'Start Without Debugging' doesn't work

### ISSUE THREAD

**Issue by xster** [Dart-Code/Dart-Code#366]
On 2.1.0. 

Steps:
`flutter create` a project, cd into it and `code .`. Debug->Start Debugging->Flutter mobile app works and creates launch.json in .vscode.

Debug->Start Without Debugging->Flutter mobile app throws:

```
Missing property 'type' for the chosen launch configuration.: Error: Missing property 'type' for the chosen launch configuration.
    at Object.t.create (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:28:65895)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:29:515653
    at n.Class.define.cancel.then (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:28:59590)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:29:515495
    at n.Class.derive._creator.then (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:28:59590)
    at t.createProcess (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:29:515425)
    at handler (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:29:120564)
    at t.invokeFunction (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:28:1062013)
    at t._tryExecuteCommand (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:28:1048628)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:28:1048328
e.onUnexpectedError @ shell.ts:443
(anonymous) @ shell.ts:403
e.onUnexpectedError @ errors.ts:119
o @ errors.ts:139
(anonymous) @ errors.ts:49
(anonymous) @ errors.ts:44
```

No .vscode directory or file created. 

Creating a launch.json first via Start Debugging and then doing Start Without Debugging works.

**Comment by DanTup** [comment]
@xster Thanks for tracking this down! Seems the code we copied from node debug for providing initial configuration was slightly flawed - it assumed we got 0 items in the debug config if it didn't exist, but we actually get the `noDebug` value if you're doing `Start without debugging`! Seems like it works in node debugger because there's code further down which detects and sets the `type`, but it means the behaviour is different (runs immediately) to debugging (creates the file and opens it).

I've updated ours to behave the same as for debugging; it'll pre-populate the `launch.json` and open it. In future we might tweak this though to just run immediately.
